### ISSUE TITLE
False positive when isset is being used to check whether a class prop is initialized or not

### ISSUE THREAD

**Issue by ZebulanStanphill** [phpstan/phpstan-strict-rules#124]
Given the following code:

```php
class MyClass {
	private int $cachedValue;

	public function getValue(): int {
		if (isset($this->cachedValue)) {
			return $this->cachedValue;
		}

		$this->cachedValue = 7;

		return $this->cachedValue;
	} 
}
```
PHPStan will report this as an error:

```
Property MyClass::$cachedValue (int) in isset() is not nullable.
```

But in this case, I'm not using `isset` to ensure that `$this->cachedValue` is not `null`, but rather I'm using it to ensure that it's initialized.

**Comment by ondrejmirtes** [comment]
Can't reproduce it: https://phpstan.org/r/f834772e-6646-4bcf-a9b4-f367fe34440f

Make sure you're on the latest PHPStan version.

**Comment by ZebulanStanphill** [comment]
@ondrejmirtes Sorry, I tried to simplify the example from my actual use-case, but apparently that prevented the error from happening. I've updated the description with something closer to the actual code, so it now actually shows the error in the sandbox. Please reopen.

**Comment by ondrejmirtes** [comment]
Properties without a native type do not have an uninitialized state. You can make it nullable: https://phpstan.org/r/9fc6e27e-2099-46a7-8e5a-6c876d902af7

**Comment by ZebulanStanphill** [comment]
> Properties without a native type do not have an uninitialized state.

Why not? Could you elaborate on this?

As far as I can tell, the PHP I wrote is completely valid, and the analysis by PHPStan about the conditional being unnecessary is incorrect. Please correct me if I'm wrong.

Functionally, there's a difference between allowing something to be uninitialized at first versus allowing it to be set to null at any time. Overriding the default WP stub type isn't really a solution, in my opinion, but rather a workaround... which I do appreciate, but to me it seems like a bug for PHPStan to not work with props that lack native typehints in this situation.

**Comment by dktapps** [comment]
Properties without a type have an implicit value of `null`, so they aren't technically uninitialized (at least not by 7.4 typed property standards).

**Comment by ZebulanStanphill** [comment]
@dktapps Oh, interesting. But the `isset` check is still valid in either case, isn't it?
